manager/scheduler/filter.go

```diff
@@ -246,13 +246,28 @@ func (f *PlatformFilter) Check(n *NodeInfo) bool {
 	// check if the platform for the node is supported
 	nodePlatform := n.Description.Platform
 	for _, p := range f.supportedPlatforms {
-		if (p.Architecture == "" || p.Architecture == nodePlatform.Architecture) && (p.OS == "" || p.OS == nodePlatform.OS) {
+		if f.platformComparison(*p, *nodePlatform) {
```

Comment:
Unrelated to the change being made here, but I think there needs to be a check that `n.Description` is not nil before using it, and also a check that `n.Description.Platform` is not nil.